### DISCUSSION TITLE
subscription model and questions about how twin basic could fit into my work flow

### DISCUSSION THREAD

**Discussion by michaelleewebb** [twinbasic/twinbasic#1743] [General]
i'm sold, i've been trying out twin basic for a few weeks now, very kool. i'm looking at the subscription model and i have questions about how i would go about using it within my working framework. i hope this is the correct place to ask these questions if not kindly tell me where to go. here's how things go for me:
i've been writing an msaccess program now since about 1998, it will never be completed, i have 4 or 5 people who use the program everyday. i have one other person that helps me with the GUI stuff mostly. what i'm writing in twin basic is not for the users, its for me and the person that helps me. right now i'm mostly working on a version control system but in the future i want twin basic to help me with other things. the program i started off with is your sample 5 com-addin program. i've modified it so that its nearly ready to be used.

so can i use twin basic like this: i have the professional edition, i make changes to the twin basic dll, i copy the dll to (somewhere) on the other msaccess development computer,  making sure i keep the dll name the same and i always put the dll in the same place. this other msaccess development computer will see the menu items i have created in the vbide. when i make updates to the msaccess program the users will not see the menu items i have created in twin basic nor will the msaccess access program complain about a "missing reference".

okay let me try to explain another way, 99.9 percent of the time i send out new version of the msaccess program to everyone automatically. i want everyone to see and use things like they do now. except for one other development computer, i want it to see the twin basic dll and autoload it and be able to use its features.

00.1 percent of the time i make changes to the twin basic dll that i have created, i want to be able to pass those changes to another development (same as above) computer easily.
can i do this ?

**Comment by fafalone** [comment]
I'm not entirely sure if this is a technical question or a license terms question... I'm leaning towards the latter but if I'm wrong let me know:

There's no license restrictions on distributing binaries made with twinBASIC; you can put your compiled DLL on as many systems as you want and no royalties are owed, including for commercial use (which is permitted), under the current terms.

For paid tiers, the twinBASIC IDE itself is only to be installed on up to 2 development machines, for use one at a time for a single user per license.

Source: https://twinbasic.com/preorder.html

**Comment by michaelleewebb** [comment]
I guess mostly the thing I want to know is what's the best way to deploy a
twinbasic dll to another computer ? I would think best is to install twin
basic on target machine. load source code for dll. compile. after that when
i have updates i want to deploy on target machine i could just copy the dll
to the target machine and overwrite the old dll.

**Reply by Greedquest** [reply]
DLLs for COM libraries need installing to register the classes. Vbide addins need registration in the registry to tell vbide to load then. I think the twin basic compiler does this for you when you compile. However if the classes in your library change in any way you'll need to re-register. If it's just a vbide addin you may get away with overwriting the dll in-place to upgrade your addin.

You can use a dll declare to point to a standard dll - and even the loadlibrary API to hot reload it and swap out DLLs without registering.

I also think there is a kind of COM dll that doesn't need to be registered - so you download from the internet and add a reference to it and it just works, but I don't think tB can compile this kind yet.

**Reply by fafalone** [reply]
You can register without compiling... regsvr32.exe with the DLL path.

Use the one in System32 for 64bit, SysWOW64 for 32bit. 

Can unregister with a switch for the same program.
